medmodels/treatment_effect_estimation/tests/test_treatment_effect.py

```diff
@@ -0,0 +1,644 @@
+            .finish()
+        )
+
+        self.assertEqual(tee2._outcome_before_treatment_days, 30)
+
+        treated_group = tee2._find_treated_patients(self.medrecord)
+        treated_group, treatment_true, outcome_before_treatment_nodes = (
+            tee2._find_outcomes(self.medrecord, treated_group)
+        )
+        self.assertEqual(treated_group, set({"P2", "P6"}))
+        self.assertEqual(treatment_true, set({"P2"}))
+        self.assertEqual(outcome_before_treatment_nodes, set({"P3"}))
+
+    def test_filter_controls(self):
+        tee = (
+            TreatmentEffect.builder()
+            .set_treatment("Rivaroxaban")
+            .set_outcome("Stroke")
+            .filter_controls(
+                node().has_outgoing_edge_with(edge().connected_target("M2"))
+                | node().has_incoming_edge_with(edge().connected_source("M2"))
+            )
+            .finish()
+        )
+        counts_test = {
+            "treatment_true": 2,
+            "treatment_false": 1,
+            "control_true": 1,
+            "control_false": 2,
+        }
+        counts_tee = tee.estimate.subject_counts(self.medrecord)
+
+        self.assertDictEqual(counts_tee, counts_test)
+
+        # filter females only
+
+        tee2 = (
+            TreatmentEffect.builder()
+            .set_treatment("Rivaroxaban")
+            .set_outcome("Stroke")
+            .filter_controls(node().attribute("gender").equal("female"))
+            .finish()
+        )
+
+        counts_test2 = {
+            "treatment_true": 2,
+            "treatment_false": 1,
+            "control_true": 1,
+            "control_false": 1,
+        }
+        counts_tee2 = tee2.estimate.subject_counts(self.medrecord)
+
+        self.assertDictEqual(counts_tee2, counts_test2)
+
+    def test_nearest_neighbors(self):
+        tee = (
+            TreatmentEffect.builder()
+            .set_treatment("Rivaroxaban")
+            .set_outcome("Stroke")
+            .adjust_with_nearest_neighbors_matching(distance_metric="absolute")
+            .finish()
+        )
+
+        subjects = tee.estimate.subjects_treatment_control(self.medrecord)
+
+        # multiple patients are equally similar to the treatment group
+        # these are exact macthes and should always be included
+        self.assertIn("P4", subjects["control_true"])
+        self.assertIn("P5", subjects["control_false"])
+
+    def test_repeat_test(self):
+        n_tests = 30
+        for n in range(n_tests):
+            print("--------------------------------")
+            print(f"Test {n} out of {n_tests}")
+
+    def test_metrics(self):
+        """Test the metrics of the TreatmentEffect class."""
+
+        tee = (
+            TreatmentEffect.builder()
+            .set_treatment("Rivaroxaban")
+            .set_outcome("Stroke")
+            .finish()
+        )
+
+        # Calculate metrics
+        self.assertAlmostEqual(tee.estimate.absolute_risk(self.medrecord), 1 / 6)
```

Comment:
Missing the cases in which there are errors raised for these metrics. They are very difficult to replicate one by one so maybe that can be skipped? @JabobKrauskopf

Reply:
fine by me :)